src/line_opacity.jl

```diff
@@ -1,28 +1,80 @@
 """
-    line_profile(temp, atomic_mass, line, wl)
+    line_opacity(linelist, wls, temp, n_density, abundances; window_size)
+
+Calculate the opacity coefficient, α, from all lines in `linelist`, at wavelengths `wls`.
+
+other arguments:
+- `temp` the temerature in K
+- `window_size` (optional, default: 20), the maximum distance from the line center at which line 
+opacities should be calculated in included.
+- `vmic` is the microturbulent velocity in cm/s
+"""
+function line_opacity(linelist, wls, temp, n_densities::Dict, atomic_masses::Dict, 
+                      partition_fns::Dict, ionization_energies::Dict, vmic
+                      ; window_size=20.0)
+
+    α_lines = zeros(length(wls))
+    for line in linelist
+        mask = line.wl - window_size .< wls .< line.wl + window_size
+
+        #line profile (normalized)
+        ϕ = line_profile(temp, atomic_masses[get_elem(line.species)], vmic, line, wls[mask])
+
+        #cross section
+        σ = sigma_line(line.wl, line.log_gf)
+
+        #stat mech quatities
+        #number density of particles in the relevant excitation state
+        boltzmann_factor = exp(- line.E_lower / kboltz_eV / temp)
+        n = n_densities[line.species] * boltzmann_factor / partition_fns[line.species](temp)
+        #the factor (1 - exp(hν₀ / kT))
+        levels_factor = 1 - exp(-c_cgs * hplanck_cgs / (line.wl * 1e-8) / kboltz_cgs / temp)
+
+        α_lines[mask] += ϕ * σ * n * levels_factor 
+    end
+    α_lines
+end
+
+"""
+    sigma_line(line, wls)
+
+The cross-section of `line`.
+"""
+function sigma_line(wl, log_gf) where F <: AbstractFloat
+    #work in cgs
+    λ = wl .* 1e-8 
+    e  = electron_charge_cgs
+    mₑ = electron_mass_cgs
+    c  = c_cgs
+
+    (π*e^2/mₑ/c) * (λ^2/c) * 10^log_gf 
+end
+
+"""
+    line_profile(temp, atomic_mass, ξ, line, wl)
 
 The line profile, ϕ, at wavelengths `wls` in Ångstroms.
-`temp` should be K, `atomic_mass` should be in g.
+`temp` should be K, `atomic_mass` should be in g, `ξ` should be in cm/s,
 `line` should be one of the entries returned by `read_line_list`.
+Note that this returns values in units of cm^-1, not Å^-1
 """
-function line_profile(temp::F, atomic_mass::F, line::NamedTuple, wls::AbstractVector{F}
+function line_profile(temp::F, atomic_mass::F, ξ, line::NamedTuple, wls::AbstractVector{F}
                      ) where F <:  AbstractFloat
     #work in cgs
     λs = wls .* 1e-8 
     λ0 = line.wl * 1e-8
 
     #doppler-broadening parameter
-    Δλ_D = λ0 * sqrt(2kboltz_cgs*temp / atomic_mass) / c_cgs
+    Δλ_D = λ0 * sqrt(2kboltz_cgs*temp / atomic_mass + ξ^2) / c_cgs
```

Comment:
It does! https://docs.julialang.org/en/v1/manual/integers-and-floating-point-numbers/#man-numeric-literal-coefficients

Note that it modifies the operator precedence.  This sounds like it would cause bugs, but it hasn't happened to me in practice.